src/mono/wasm/host/DevServer/ComponentsWebAssemblyApplicationBuilderExtensions.cs

```diff
@@ -0,0 +1,138 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.StaticFiles;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.FileProviders;
+using Microsoft.Net.Http.Headers;
+using System;
+using System.IO;
+using System.Net.Mime;
+
+namespace Microsoft.WebAssembly.AppHost.DevServer;
+
+internal static class ComponentsWebAssemblyApplicationBuilderExtensions
+{
+    private static readonly string? s_dotnetModifiableAssemblies = GetNonEmptyEnvironmentVariableValue("DOTNET_MODIFIABLE_ASSEMBLIES");
+    private static readonly string? s_aspnetcoreBrowserTools = GetNonEmptyEnvironmentVariableValue("__ASPNETCORE_BROWSER_TOOLS");
+
+    private static string? GetNonEmptyEnvironmentVariableValue(string name)
+        => Environment.GetEnvironmentVariable(name) is { Length: > 0 } value ? value : null;
+
+    /// <summary>
+    /// Configures the application to serve Blazor WebAssembly framework files from the path <paramref name="pathPrefix"/>. This path must correspond to a referenced Blazor WebAssembly application project.
+    /// </summary>
+    /// <param name="builder">The <see cref="IApplicationBuilder"/>.</param>
+    /// <param name="pathPrefix">The <see cref="PathString"/> that indicates the prefix for the Blazor WebAssembly application.</param>
+    /// <returns>The <see cref="IApplicationBuilder"/></returns>
+    public static IApplicationBuilder UseBlazorFrameworkFiles(this IApplicationBuilder builder, PathString pathPrefix)
+    {
+        ArgumentNullException.ThrowIfNull(builder);
+
+        var webHostEnvironment = builder.ApplicationServices.GetRequiredService<IWebHostEnvironment>();
+
+        var options = CreateStaticFilesOptions(webHostEnvironment.WebRootFileProvider);
+
+        builder.MapWhen(
+            ctx => ctx.Request.Path.StartsWithSegments(pathPrefix, out var rest)
+                && rest.StartsWithSegments("/_framework")
+                && !rest.StartsWithSegments("/_framework/blazor.server.js")
+                && !rest.StartsWithSegments("/_framework/blazor.web.js"),
```

Comment:
What we need here to make non-blazor also hot-reload enabled ?

Reply:
There is more work on that one and I don't know all the details yet. There is a module in SDK https://github.com/dotnet/sdk/blob/13b8781fe194a26bc51f1bf766a2f85c04e6298a/src/BuiltInTools/BrowserRefresh/BlazorHotReload.js which is part of the aspnetcore browser reload tools and same bits in Blazor https://github.com/dotnet/aspnetcore/blob/6a93078ee02d3e2aa8ab196ae8911e8bdb803a7b/src/Components/WebAssembly/WebAssembly/src/HotReload/WebAssemblyHotReload.cs#L45